src/components/NewCommentForm.tsx

```diff
@@ -2,98 +1,194 @@
+    e.preventDefault();
+
+    const isNameEmpty = newComment.name === '';
+    const isEmailEmpty = newComment.email === '';
+    const isBodyEmpty = newComment.body === '';
+
+    if (isNameEmpty || isEmailEmpty || isBodyEmpty) {
+      setTouched({
+        name: isNameEmpty,
+        email: isEmailEmpty,
+        body: isBodyEmpty,
+      });
+
+      return;
+    }
+
+    try {
+      await onAdd(newComment);
+
+      setNewComment(prevComment => ({ ...prevComment, body: '' }));
+      setTouched({ name: false, email: false, body: false });
+    } catch {
+      setTouched({ name: true, email: true, body: true });
+    }
+  };
 
-export const NewCommentForm: React.FC = () => {
   return (
-    <form data-cy="NewCommentForm">
+    <form data-cy="NewCommentForm" onSubmit={handleSubmit}>
       <div className="field" data-cy="NameField">
         <label className="label" htmlFor="comment-author-name">
           Author Name
         </label>
-
         <div className="control has-icons-left has-icons-right">
           <input
             type="text"
             name="name"
             id="comment-author-name"
             placeholder="Name Surname"
-            className="input is-danger"
+            value={newComment.name}
+            onChange={handleInputChange}
+            className={cn('input', {
+              'is-danger': touched.name && newComment.name === '',
+            })}
           />
-
           <span className="icon is-small is-left">
             <i className="fas fa-user" />
           </span>
-
-          <span
-            className="icon is-small is-right has-text-danger"
-            data-cy="ErrorIcon"
-          >
-            <i className="fas fa-exclamation-triangle" />
-          </span>
         </div>
 
-        <p className="help is-danger" data-cy="ErrorMessage">
-          Name is required
-        </p>
+        {touched.name && newComment.name === '' && (
+          <>
+            <span
+              className="icon is-small is-right has-text-danger"
+              data-cy="ErrorIcon"
+            >
+              <i className="fas fa-exclamation-triangle" />
+            </span>
+            <p className="help is-danger" data-cy="ErrorMessage">
+              Name is required
+            </p>
+          </>
+        )}
       </div>
 
       <div className="field" data-cy="EmailField">
         <label className="label" htmlFor="comment-author-email">
           Author Email
         </label>
-
         <div className="control has-icons-left has-icons-right">
           <input
-            type="text"
+            type="email"
             name="email"
             id="comment-author-email"
             placeholder="email@test.com"
-            className="input is-danger"
+            value={newComment.email}
+            onChange={handleInputChange}
+            className={cn('input', {
+              'is-danger': touched.email && newComment.email === '',
```

Comment:
Ensure that the email input field is validated for proper email format. Currently, it only checks if the field is empty. Consider using a regex pattern to validate the email format.